validation/standard/standard-version-consistency_test.go

```diff
@@ -0,0 +1,74 @@
+package standard_test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum-optimism/superchain-registry/validation"
+	"github.com/ethereum-optimism/superchain-registry/validation/standard"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+// This test will check:
+// For each each superchain
+// For each contract release
+// For each contract version declaration which specifies an address
+// There is a contract at that address with the matching bytecode and semver.
+// This is a consistency check on the standard package itself, not any particular chain
+func TestStandardVersionConsistency(t *testing.T) {
+	for _, superchain := range []string{"sepolia", "mainnet"} {
+		rpcEndpoint := Superchains[superchain].Config.L1.PublicRPC
+		require.NotEmpty(t, rpcEndpoint)
+
+		client, err := ethclient.Dial(rpcEndpoint)
+		require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+		for tag, release := range standard.NetworkVersions[superchain].Releases {
+			for _, contract := range release.GetNonEmpty() {
```

Comment:
In general it seems like this `GetNonEmpty` method is dangerous because we don't actually know what we're checking for any given chain/release. Could we define some clear definitions of valid contract sets (e.g. known expected contracts for [1] permissioned fault-proofs, [2] permissionless fault-proofs [3] non-fault proofs) ? Or are there too many combinations of contract sets to make that feasible?

Reply:
Similar function used for `L1ContractBytecodeHashes.GetNonEmpty()`

Reply:
Can address this in a separate pr since its a large refactor outside the scope of this work